From 10e1f5efd880cec04616558b31a3557448e61e29 Mon Sep 17 00:00:00 2001
From: Jerome Robert <jeromerobert@users.sf.net>
Date: Thu, 10 Apr 2014 18:31:38 +0200
Subject: [PATCH] ApproxAFunc2Var: numerical bug triggered by compiler
 optimization

Here Result and Constraints UpdateInX method expect exactly the same value.
Because of the compiler optimization the evaluation of
myFirstParInU+iint*deltu may be different between Result and Constraints
update call.
---
 src/AdvApp2Var/AdvApp2Var_ApproxAFunc2Var.cxx | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

Index: oce/src/AdvApp2Var/AdvApp2Var_ApproxAFunc2Var.cxx
===================================================================
--- oce.orig/src/AdvApp2Var/AdvApp2Var_ApproxAFunc2Var.cxx
+++ oce/src/AdvApp2Var/AdvApp2Var_ApproxAFunc2Var.cxx
@@ -293,10 +293,12 @@ void AdvApp2Var_ApproxAFunc2Var::InitGri
   Standard_Real deltu = (myLastParInU-myFirstParInU)/NbInt,
                 deltv = (myLastParInV-myFirstParInV)/NbInt;
   for (iint=1;iint<=NbInt-1;iint++) {
-    Result.UpdateInU(myFirstParInU+iint*deltu);
-    Constraints.UpdateInU(myFirstParInU+iint*deltu);
-    Result.UpdateInV(myFirstParInV+iint*deltv);
-    Constraints.UpdateInV(myFirstParInV+iint*deltv);
+    Standard_Real u = myFirstParInU + iint * deltu;
+    Standard_Real v = myFirstParInV + iint * deltv;
+    Result.UpdateInU(u);
+    Constraints.UpdateInU(u);
+    Result.UpdateInV(v);
+    Constraints.UpdateInV(v);
   }
   myResult = Result;
   myConstraints = Constraints;
